include/beast/websocket/detail/decorator.hpp

```diff
@@ -40,6 +41,32 @@ class decorator : public abstract_decorator
 {
     T t_;
 
+    class call_req_possible
+    {
+        template<class U, class R = decltype(
+            std::declval<U>().operator()(
+                std::declval<request_type&>()),
+                    std::true_type{})>
+        static R check(int);
+        template<class>
+        static std::false_type check(...);
+    public:
+        using type = decltype(check<T>(0));
+    };
+
+    class call_res_possible
+    {
+        template<class U, class R = decltype(
+            std::declval<U>().operator()(
+                std::declval<response_type&>()),
+                    std::true_type{})>
+        static R check(int);
+        template<class>
+        static std::false_type check(...);
+    public:
+        using type = decltype(check<T>(0));
+    };
+
```

Comment:
Can `detail::is_call_possible` be reused here?  Haven't tested but it looks like `call_res_possible == is_call_possible<T, void(response_type&)>` (or something along those lines).